lib/fog/vsphere/models/compute/snapshots.rb

```diff
@@ -0,0 +1,31 @@
+require 'fog/core/collection'
+require 'fog/vsphere/models/compute/snapshot'
+
+module Fog
+  module Compute
+    class Vsphere
+      class Snapshots < Fog::Collection
+        attribute :server_id, alias: :instance_uuid
+        attribute :parent_snapshot
+        model Fog::Compute::Vsphere::Snapshot
+
+        def all(filters = {})
+          if parent_snapshot
+            load service.list_child_snapshots(parent_snapshot, filters)
+          else
+            requires :server_id
+            load service.list_vm_snapshots(server_id, filters)
+          end
+        end
+
+        def get(snapshost_ref)
+          all.each do |s|
+            nested = s.get(snapshost_ref)
+            return nested if nested
+          end
+          nil
+        end
```

Comment:
Unfortunately there is not. It is necessary to go recursively by calling [childSnapshotList](https://pubs.vmware.com/vsphere-60/index.jsp#com.vmware.wssdk.apiref.doc/vim.vm.SnapshotTree.html?resultof=%2522%2576%2569%2572%2574%2575%2561%256c%254d%2561%2563%2568%2569%256e%2565%2553%256e%2561%2570%2573%2568%256f%2574%2554%2572%2565%2565%2522%2520%2522%2576%2569%2572%2574%2575%2561%256c%256d%2561%2563%2568%2569%256e%2565%2573%256e%2561%2570%2573%2568%256f%2574%2574%2572%2565%2522%2520) method.